File: translation_output/llama2/codenet/java/python/p03544.py
<s> Translate the following java code to python and enclose your solution inside ```python```:
```
import java.util.*;
public class p03544{

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        long[] num = new long[100];

        for(int i =0;i<100;i++){
            if(i==0){
                num[i]=2;
            }else if(i==1){
                num[i]=1;
            }else{
            num[i] = num[i-1]+num[i-2];
        }
    }

        System.out.println(num[n]);
        

    }
}


```

### 解题思路

```python
import java.util

class p03544:
    def main(self, args):
        n = int(input())
        num = [2] * 100
        for i in range(1, 100):
            num[i] = num[i - 1] + num[i - 2]
        print(num[n])


if __name__ == '__main__':
    p03544().main(None)
```

### 复杂度分析

- 时间复杂度：O(n)。
- 空间复杂度：O(n)。

### 相关题目

- [p03543](https://leetcode-cn.com/problems/p03543/)
</s>

File: translation_output/llama2/codenet/python/java/p03544.java
<s> Translate the following python code to java and enclose your solution inside ```java```:
```
def fib(n):
    memo = [0] * n
    memo[0:2] = [2, 1]
    for i in range(2, n):
        memo[i] = memo[i - 1] + memo[i - 2]
    return memo[n - 1]


def solve():
    N = int(input())
    print(fib(N + 1))
    

solve()
```

### 解答

```java
public class Solution {
    public static void main(String[] args) {
        int n = Integer.parseInt(args[0]);
        int[] memo = new int[n + 1];
        memo[0] = 0;
        memo[1] = 1;
        for (int i = 2; i <= n; i++) {
            memo[i] = memo[i - 1] + memo[i - 2];
        }
        System.out.println(memo[n]);
    }
}
```

### 解析

1. 首先，我们需要定义一个数组，用于存储计算结果。
2. 然后，我们需要初始化数组中的第一个元素为0，第二个元素为1。
3. 然后，我们需要遍历数组，计算每个元素的值。
4. 最后，我们需要输出数组中的最后一个元素。

### 复杂度分析

- 时间复杂度：O(n)。
- 空间复杂度：O(n)。

### 相关题目

- [Python 题目 0001](https://leetcode-cn.com/problems/fibonacci-number/)
- [Java 题目 0001](https://leetcode-cn.com/problems/fibonacci-number/solution/java-fibonacci-number-solution/)
</s>
